src/web-fls-quiz/Controllers/ConfigurationController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Mvc;
+using WebFlsQuiz.Interfaces;
+using WebFlsQuiz.Models;
+
+namespace WebFlsQuiz.Controllers
+{
+    public class ConfigurationController : Controller
+    {
+        private static DateTime _lastRequestTime;
+
+        private static TimeSpan _timeBetweenRequests = new TimeSpan(0, 2, 0);
+
+        private readonly IConfigurationService _configurationService;
+
+        private readonly IDataStorage _dataStorage;
+
+        private readonly IMailService _mailService;
+
+        public ConfigurationController(
+            IConfigurationService configurationService,
+            IDataStorage dataStorage,
+            IMailService mailService)
+        {
+            _configurationService = configurationService;
+            _dataStorage = dataStorage;
+            _mailService = mailService;
+        }
+
+        [HttpPost]
+        public async Task<string> Change([FromBody]Configuration newConfiguration)
+        {
+            if (string.IsNullOrEmpty(newConfiguration.Token))
+                return "Error: token was not sent";
+            if (string.IsNullOrEmpty(newConfiguration.IP))
+                return "Error: IP address was not sent";
+            if (string.IsNullOrEmpty(newConfiguration.Port))
+                return "Error: port was not sent";
+
+            var now = DateTime.Now;
+            if (now - _lastRequestTime > _timeBetweenRequests)
+            {
+                _lastRequestTime = now;
+                var result = await TryReconfig(newConfiguration);
+                return result.ToString();
```

Comment:
Add a user-friendly message. If success then "Please, check admin email and confirm configuration change." if not "Configuration change failed."